### PR TITLE
SEO

### LINKED ISSUE #24: SEO support for meta, structured data, sitemap.xml, etc.
Add general SEO settings required for media sites.

### PR DESCRIPTION
**Pull request by VortexExpansion**
Fixes #24 

## Reviewer 
@HisashiHasebe @yabe-diverta 

## Checklist
- [x] Added metadata for article/privacy/detail pages https://diverta.gyazo.com/39ee1f2b7f549d788d137c16c792ddc4
- [x] sitemap https://diverta.gyazo.com/1bd460fe4ee64f3a63283f3cd7892ccd
- [x] robots.txt

### CODE REVIEW

**Comment by VortexExpansion** [comment]
@yabe-diverta san 
For sitemap there still seems to be a problem with `output:export` in Next JS 14 
Currently it works if you try to access sitemap in build but fails in dev. 
Reference : https://www.github.com/vercel/next.js/issues/59136

Shall we leave it as it is then or look for workarounds ?

**Comment by VortexExpansion** [comment]
@yabe-diverta san
Secondly, metadata cannot be exported in client components directly. 
The workaround is moving all the client components of a page out into a separate component and making all pages server side. Like what we did in articles page. https://diverta.gyazo.com/09d89b0b5b219f610d173972f4a65138

What do you suggest ? Do we do this for all the remaining pages ? Would be a big migration update. 
Or let it be and the existing metadata is sufficient.

**Comment by yabe-diverta** [comment]
@VortexExpansion 
https://github.com/diverta/front_next_media/pull/32#issuecomment-2112069584
Do you mean we can see the sitemap over deployed website but not in local?

https://github.com/diverta/front_next_media/pull/32#issuecomment-2112087020
Sitemap is for SEO,
its not necessary to list those limited pages.
Let me check we dont have limited pages which dont need logged in, right?

**Comment by VortexExpansion** [comment]
@yabe-diverta san

> Do you mean we can see the sitemap over deployed website but not in local?

Yes. Is that OK ? 

> its not necessary to list those limited pages.
> Let me check we dont have limited pages which dont need logged in, right?

Yes, mostly limited pages requiring login are client side. But apart from that certain pages like reminder, register, login, contact are also not covered by metadata currently.

**Comment by yabe-diverta** [comment]
@VortexExpansion 

>> Do you mean we can see the sitemap over deployed website but not in local?
> Yes. Is that OK ?

Yes I think so.

> Yes, mostly limited pages requiring login are client side. But apart from that certain pages like reminder, register, login, contact are also not covered by metadata currently.

That pages are supposed to be dynamically generated? 🤔 
I'm unsure why that happens.
I think they should be statically generated and meta datas filled.
Additionally if those pages were client component, you could add layout.tsx to generate meta for them.

And plus, dynamic pages other than them should have meta datas too.
Not statically but dynamically.
There is no way to change them along with Next API so we could use native JS instead.
`document.querySelector("meta[name='description']").setAttribute("content", `${value}`);`
Although changing meta in client component mostly no meaning, just for decorating page info, it's least priority to proceed I think.
You could separate to an another issue to do it.

**Comment by VortexExpansion** [comment]
@abe san as discussed,

I will deal with the metadata for client components in a separate issue marked as low priority for now. 
Meanwhile, this PR is ready for merge. Please review.

File: src/app/robots.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default function robots() {
+    return {
+      rules: {
+        userAgent: '*',
+        allow: '/',
+        // disallow: '/private/',
```
Comment (yabe-diverta) [inline src/app/robots.js]: remove?

**Review by yabe-diverta — APPROVED** [review]
pls do just one minor issue, seems fine other than that. 🙇

File: src/app/robots.js (hunk 1 above)
Comment (VortexExpansion) [reply src/app/robots.js]: Done

**Comment by VortexExpansion** [comment]
Ready for merge

**Pull request merged**